src/interfaces/dnmdowner.hpp

```diff
@@ -0,0 +1,100 @@
+#ifndef _SRC_INTERFACES_DNMDOWNER_HPP_
+#define _SRC_INTERFACES_DNMDOWNER_HPP_
+
+#include "internal/dnmd_platform.hpp"
+#include "tearoffbase.hpp"
+#include "controllingiunknown.hpp"
+
+#include <external/cor.h>
+#include <external/corhdr.h>
+
+#include <cstdint>
+#include <atomic>
+
+EXTERN_GUID(IID_IDNMDOwner, 0x250ebc02, 0x1a92, 0x4638, 0xaa, 0x6c, 0x3d, 0x0f, 0x98, 0xb3, 0xa6, 0xfb);
+
+// This interface is an IUnknown interface for the purposes of easy discovery.
+struct IDNMDOwner : IUnknown
+{
+    virtual mdhandle_t MetaData() = 0;
+};
+
+// We use a reference wrapper around the handle to allow the handle to be swapped out.
+// We plan to use swapping to implement table sorting as DNMD itself does not support
+// sorting tables or remapping tokens.
+class mdhandle_view final
+{
+private:
+    dncp::com_ptr<IDNMDOwner> owner;
+public:
+    mdhandle_view(dncp::com_ptr<IDNMDOwner> owner)
+        :owner(owner.p)
+    {
+    }
+
+    mdhandle_view(mdhandle_view const& other)
+        :owner(other.owner.p)
```

Comment:
```suggestion
        : owner{ owner.p }
    {
    }

    mdhandle_view(mdhandle_view const& other)
        : owner{ other.owner.p }
```

Reply:
Also, why copy as opposed to move?

Reply:
We want a copy here as we will pass one copy to the Import implementation's constructor and one to the Emit implementation's constructor. I can add a move constructor as well for completeness, but we definitely need a copy constructor as well.

Reply:
Nope. If we don't need it that is fine.